src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/OutputFlowControl.cs

```diff
@@ -37,6 +39,12 @@ public OutputFlowControlAwaitable AvailabilityAwaitable
             }
         }
 
+        public void Reset()
+        {
+            _flow = new FlowControl(_initialWindowSize);
+            _awaitableQueue?.Clear();
```

Comment:
If the _awaitableQueue wasn't already clear, there's a problem.

Reply:
What do you want to change this to? Debug.Assert that it is empty?

Reply:
Yeah. If you're sure that's an invariant when Reset is called.

Reply:
Added assert